language/move-prover/stackless-bytecode-generator/src/eliminate_imm_refs.rs

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    function_target::{FunctionTarget, FunctionTargetData},
+    function_target_pipeline::{FunctionTargetProcessor, FunctionTargetsHolder},
+    stackless_bytecode::{
+        AssignKind,
+        Bytecode::{self, *},
+        Operation::*,
+    },
+};
+use spec_lang::{env::FunctionEnv, ty::Type};
+
+pub struct EliminateImmRefsProcessor {}
+
+impl EliminateImmRefsProcessor {
+    pub fn new() -> Box<Self> {
+        Box::new(EliminateImmRefsProcessor {})
+    }
+}
+
+impl FunctionTargetProcessor for EliminateImmRefsProcessor {
+    fn process(
+        &self,
+        _targets: &mut FunctionTargetsHolder,
+        func_env: &FunctionEnv<'_>,
+        mut data: FunctionTargetData,
+    ) -> FunctionTargetData {
+        let func_target = FunctionTarget::new(func_env, &data);
+        data.code = data
+            .code
+            .iter()
+            .map(|bytecode| EliminateImmRefs::transform_bytecode(&func_target, bytecode))
+            .collect();
+        data.local_types = data
+            .local_types
+            .into_iter()
+            .map(EliminateImmRefs::transform_type)
+            .collect();
+        data.return_types = data
+            .return_types
+            .into_iter()
+            .map(EliminateImmRefs::transform_type)
+            .collect();
+        data
+    }
+}
+
+pub struct EliminateImmRefs {}
+
+impl EliminateImmRefs {
+    pub fn transform_type(ty: Type) -> Type {
```

Comment:
nit: perhaps make these functions take `&self` so if you ever want state in `EliminateImmRefs` its an easy change. Or you can put FunctionTarget into here and don't need to pass it.

Reply:
Done.